cli/src/main.rs

```diff
@@ -105,6 +105,7 @@ struct Cli {
 #[derive(Subcommand)]
 enum Commands {
     /// Runs compilation and witness generation for .pil and .asm files.
+    /// Also runs backend if `--prove-with` is set.
```

Comment:
What does `runs backend` mean? Generates a proof? Also maybe put this at the end (such that it is in sequence of operations)